frontend/javascripts/admin/task/task_create_form_view.js

```diff
@@ -191,14 +193,15 @@ class TaskCreateFormView extends React.PureComponent<Props, State> {
   }
 
   async fetchData() {
+    this.setState({ isFetchingData: true });
     const [datasets, projects, scripts, taskTypes] = await Promise.all([
       getActiveDatasets(),
       getProjects(),
       getScripts(),
       getTaskTypes(),
     ]);
-
     this.setState({ datasets, projects, scripts, taskTypes });
+    this.setState({ isFetchingData: false });
```

Comment:
You should be able to combine these two setState calls into one, simply add the `isFetchingData: false` in the first call